src/portable/portable.c

```diff
@@ -994,15 +1058,30 @@ static int install_chroot_dropin(
                                "LogExtraFields=PORTABLE=", base_name, "\n"))
                         return -ENOMEM;
 
+                /* If we have a single image then PORTABLE= will point to it, so we add PORTABLE_ID= with
+                 * the os-release fields and we are done. But if we have extensions, PORTABLE= will point
+                 * to the image where the current unit was found in. So we also list PORTABLE_ROOT= and
+                 * PORTABLE_ROOT_ID= for the base image, and PORTABLE_EXTENSION= and PORTABLE_EXTENSION_ID=
+                 * for each extension, so that all needed metadata is available. */
                 if (!ordered_hashmap_isempty(extension_images)) {
                         _cleanup_free_ char *root_base_name = NULL;
 
                         r = path_extract_filename(image_path, &root_base_name);
                         if (r < 0)
                                 return log_debug_errno(r, "Failed to extract basename from '%s': %m", image_path);
 
-                        if (!strextend(&text, "LogExtraFields=PORTABLE_ROOT=", root_base_name, "\n"))
+                        if (!strextend(&text,
+                                       "Environment=PORTABLE_ROOT=", root_base_name, "\n",
+                                       "LogExtraFields=PORTABLE_ROOT=", root_base_name, "\n"))
                                 return -ENOMEM;
+
+                        r = append_release_log_fields(os_release, IMAGE_PORTABLE, "PORTABLE_ROOT_ID", &text);
+                        if (r < 0)
+                                return r;
+                } else {
+                        r = append_release_log_fields(os_release, IMAGE_PORTABLE, "PORTABLE_ID", &text);
+                        if (r < 0)
+                                return r;
```

Comment:
the last two lines of both if branches are the same. move out of the if branches

Reply:
updated